src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java

```diff
@@ -104,24 +117,58 @@ public static VerificationCheck checkSmsVerificationCode(String phoneNumber, Str
         }
     }
 
+    /**
+     * Send notification email for {@link OtpUser}, ensuring the correct from
+     * email address is used (i.e., {@link #FROM_EMAIL}).
+     */
+    public static boolean sendEmail(OtpUser otpUser, String subject, String text, ContainerTag html) {
+        // Inject manage subscriptions tag.
+        if (html != null) html.with(manageSubscriptions(OTP_UI_URL + "/#/account"));
+        return sendEmailViaSparkpost(FROM_EMAIL, otpUser.email, subject, text, html);
+    }
+
+    /**
+     * Send notification email for {@link AdminUser}, ensuring the correct from
+     * email address is used (i.e., {@link #OTP_ADMIN_DASHBOARD_EMAIL}).
+     */
+    public static boolean sendEmail(AdminUser adminUser, String subject, String text, ContainerTag html) {
+        // Inject manage subscriptions tag.
+        if (html != null) html.with(manageSubscriptions(OTP_ADMIN_DASHBOARD_URL + "/account"));
+        return sendEmailViaSparkpost(OTP_ADMIN_DASHBOARD_EMAIL, adminUser.email, subject, text, html);
+    }
+
+    /**
+     * @return A "Manage my subscriptions" tag for email notifications.
+     */
+    public static ContainerTag manageSubscriptions(String url) {
+        return p(small(join("Manage subscriptions", a("here").withHref(url), ".")));
```

Comment:
I do not like this j2html library at all for this purpose. We should use templates instead. Seems like https://freemarker.apache.org/ is the Java go-to.

Reply:
Ok, just to follow-up, this is definitely a knee-jerk reaction to seeing something so different than using templates. I read up on this library and it seems like it is very much oriented towards building highly-performant code. Maybe we'll reach a point where we might need this, but there are so many tradeoffs that using this library has. Templates will be much much easier to read, edit and track in version control.